tests/unit/CurrencyUtilsTest.js

```diff
@@ -89,7 +89,9 @@ describe('CurrencyUtils', () => {
         test.each([
             [CONST.CURRENCY.USD, 25, 2500],
             [CONST.CURRENCY.USD, 25.5, 2550],
-            [CONST.CURRENCY.USD, 25.5, 2550],
+            [CONST.CURRENCY.USD, 80.6, 8060],
+            [CONST.CURRENCY.USD, 80.9, 8090],
+            [CONST.CURRENCY.USD, 80.99, 8099],
             ['JPY', 25, 25],
             ['JPY', 2500, 2500],
             ['JPY', 25.5, 25],
```

Comment:
I think what this test was trying to do with that doubled test was to copy the same test cases in both USD and JPY, and something just got lost along the way. Here's my suggestion here:

```
            [CONST.CURRENCY.USD, 25, 2500],
            [CONST.CURRENCY.USD, 25.5, 2550],
            [CONST.CURRENCY.USD, 2500, 250000],
            [CONST.CURRENCY.USD, 80.6, 8060],
            [CONST.CURRENCY.USD, 80.9, 8090],
            [CONST.CURRENCY.USD, 80.99, 8099],
            ['JPY', 25, 25],
            ['JPY', 2500, 2500],
            ['JPY', 25.5, 25],
            ['JPY', 80.6, 80],
            ['JPY', 80.9, 80],
            ['JPY', 80.99, 80],
```

My only question here is whether or not this would be expected behavior for the JPY cases. Would we really expect 80.99 to come in as 80? cc @roryabraham in case you happen to know. If you don't, I think we stick with it because that isn't changing any behavior from what exists now...

Reply:
Regarding JPY, somewhere in the comments this issue was linked: https://github.com/Expensify/App/issues/15878

Reply:
Ha, that is a LONG chain. I THINK, if I'm understanding it correctly, that we would actually want to round up 80.99 to 81 because that's what we'd display:

> [When we display the value, we're using the standard Intl.NumberFormat function so it's displaying 25.75 JPY correctly as 26 JPY](https://github.com/Expensify/App/issues/15878#issuecomment-1469972557)

I think most of the floor/trunc logic is only when we're splitting things, so we wouldn't want to also do it here.

But I'll cc @aldo-expensify since he worked on that issue and maybe he can confirm one way or the other - do we want to round up/down decimals in currencies without decimals, or should we always truncate/floor them?

Reply:
I'm not sure I understand everything going on here... but I would ask first why are we testing conversions of decimal JPY. Is it possible for a user to input JPY with decimal positions? if it is not, maybe then these test are testing things that can't happen.

Reply:
ahh, this is not necessarily related to currencies like JPY, but this instead:

```
Number.parseFloat('80.60') * 100
8059.999999999999
```

Reply:
I had that question too actually, just checked and yes, you can enter decimal for JPN currently.

> Unrelated to this issue but it would be better to prevent users from doing that IMO.

Reply:
I think the code changes make sense.

Regarding JPY, I wouldn't care much about that for now. I'm guessing that some code in the backend could produce JPY decimals... but 🤷

Reply:
@dangrous so should I round it or leave it?